Resources/public/js/views/services/ez-contenteditviewservice.js

```diff
@@ -24,6 +24,7 @@ YUI.add('ez-contenteditviewservice', function (Y) {
     Y.eZ.ContentEditViewService = Y.Base.create('contentEditViewService', Y.eZ.ViewService, [], {
         initializer: function () {
             this.on('*:closeView', this._handleCloseView);
+            this.on('*:requestChange', this._handleRequestChange);
```

Comment:
I would rename `_handleRequestChange` to `_setLanguageCode` or something similar, that way you directly what happen when the request change.